src/js/angular/ttyg/directives/chat-panel.directive.js

```diff
@@ -64,10 +64,21 @@ function ChatPanelComponent(toastr, $translate, TTYGContextService, TTYGService)
                 if (!$scope.chatItem.chatId) {
                     createNewChat();
                 } else {
-                    askQuestion();
+                    askQuestion($scope.chatItem);
                 }
             };
 
+            $scope.regenerate = (chatItem) => {
```

Comment:
write jsdoc for the public methods and fields at least!

Reply:
Also, can you be more specific in the methods naming.

Reply:
The name has been changed, and Javadoc has been added.